### PR TITLE
Implemented prototype of Kover artifacts

### PR DESCRIPTION
**Pull request by shanshin**

### CODE REVIEW

File: kover/kover-cli/src/main/kotlin/kotlinx/kover/cli/commands/OfflineInstrumentCommand.kt [hunk 1]
```diff
@@ -0,0 +1,67 @@
...
+
+internal class OfflineInstrumentCommand : Command {
+    // hint: MutableList used to remove variance, args4j accept java.util.List<File> but not java.util.List<? extends File>
+    @Argument(usage = "list of the compiled class-files roots", metaVar = "<class-file-path>")
+    private var roots: MutableList<File> = ArrayList()
+
+    @Option(name = "--dest", usage = "path to write instrumented Java classes to", metaVar = "<dir>", required = true)
+    private var outputDir: File? = null
+
+    @Option(name = "--hits", usage = "path to write instrumented Java classes to")
```
Comment (zuevmaxim) [inline kover/kover-cli/src/main/kotlin/kotlinx/kover/cli/commands/OfflineInstrumentCommand.kt:36]: Here usage string is unrelated. Should be `a flag to enable line hits counting`?

File: kover/kover-cli/src/main/kotlin/kotlinx/kover/cli/commands/ReportCommand.kt [hunk 2]
```diff
@@ -0,0 +1,119 @@
...
+    @Option(
+        name = "--excludeAnnotation",
+        usage = "filter to include classes and functions marked by this annotation, wildcards `*` and `?` are acceptable",
+        metaVar = "<annotation-name>"
+    )
+    private var excludeAnnotation: MutableList<String> = ArrayList()
+
+    override val name: String = "report"
+
+    override val description: String = "Generates human-readable reports in various formats from raw report files (*.ic)"
```
Comment (zuevmaxim) [inline kover/kover-cli/src/main/kotlin/kotlinx/kover/cli/commands/ReportCommand.kt:80]: In this repository, `.ic` files are named 'binary' reports, while 'raw' reports/format is related only to the results of the offline instrumented coverage run.

File: kover/kover-cli/src/main/kotlin/kotlinx/kover/cli/commands/ReportCommand.kt [hunk 3]
```diff
@@ -0,0 +1,119 @@
...
+import org.kohsuke.args4j.Argument
+import org.kohsuke.args4j.Option
+import java.io.File
+import java.io.IOException
+import java.io.PrintWriter
+
+
+internal class ReportCommand : Command {
+    @Argument(usage = "list of raw reports files", metaVar = "<raw-report-path>")
+    private var rawReports: MutableList<File> = ArrayList()
```
Comment (zuevmaxim) [inline kover/kover-cli/src/main/kotlin/kotlinx/kover/cli/commands/ReportCommand.kt:35]: `binaryReports` is a better name here, if `.ic` files are supposed to be used

**Comment by BalanceYKT** [comment]
Sorry for the intrusion, I couldn't find in the Kover documentation if it is possible to generate a report for a single file. For example, something like features:myFeature1:interactor:koverHtmlReport. Is it possible?

**Comment by shanshin** [comment]
> Sorry for the intrusion, I couldn't find in the Kover documentation if it is possible to generate a report for a single file. For example, something like features:myFeature1:interactor:koverHtmlReport. Is it possible?

Hi, these changes relate to the Kover toolkit - a set of dependencies that will be used via the command line, or in plugins for the build systems.

If you are using the Kover Gradle Plugin, then please create an issue in the [corresponding project](https://github.com/Kotlin/kotlinx-kover/issues)

**Review by zuevmaxim — APPROVED** [review]

**Pull request merged**